### ISSUE TITLE
The NG documentation website is broken

### ISSUE THREAD

**Issue by sno036** [APSIMInitiative/ApsimX#5606]
(again)

**Comment by hol430** [comment]
Sounds like a duplicate of #5258. I can fix the problem in this instance, but in order to prevent this from happening again, we would need to generate the autodocs as part of regular jenkins builds. This would significantly increase the testing time. If we don't want to do this, then we can continue to find out about breakages after the fact.

**Comment by sno036** [comment]
Yes it looks like a duplication. Could/should be merged. Is there a happier medium? 

- At the moment it seems that there is a reliance on a person reporting the breakage? Can it be automated?
- Is it a failure in one document that cascades? If so can there just be a blank or error message for the failed document?
- Can it be made more robust?

**Comment by hol353** [comment]
When I get back on CSIRO time, my first job is to move our build system to cloud based and to rethink our build scripts so that something like auto-docs breaking isn't left unnoticed.

**Comment by hol430** [comment]
If we start to change this process to make it continue despite errors, then we end up in the situation where upgrades are still being released, but some docs are missing from the documentation page. On the other hand, if we make it abort early, then we end up with a complete documentation page, but no upgrades being released until the problem is fixed. Automating the error reporting (ie sending an email on job failure) will help with this and is definitely worth doing, but I don't think it's a solution by istelf.

If and when we do change this process and remove our physical compute nodes, perhaps we can start generating autodocs as part of the main testing phase. Then we just throw more resources at the problem so it doesn't take longer. I also suspect the autodocs code is not very efficient, and could be refactored to significantly improve execution time.

**Comment by jbrider** [comment]
If the resources are free, then throwing more at it is fine, but if there is a cost then it may need some way to detect when the docs are required to be rebuilt? Or perhaps just a daily process that runs if changes have been made during the day. Keen to be involved in this process.

**Comment by hol430** [comment]
Yeah there will be a cost to throwing more resources at the problem. Rebuilding only when necessary would be an improvement, but I can't think of a way to implement this. Theoretically, any change to the code in Models could cause the docs to need to be rebuilt. Rebuilding the docs once a day could be a possibility, but that would also involve moving to a daily release cycle, since we would always want to keep the docs in sync with the latest released version. I don't really see a problem with moving to daily releases if that's what we want to do, but it doesn't solve the problem of not knowing that we've broken the docs until after the pull request gets merged.

**Comment by jbrider** [comment]
Good point about keeping it in sync. 
The reason I was looking for the docs was to check if formula's all appear the same as the pasture documentation I was looking at - it looked really good except for the small formula's that were right aligned to the page.

**Comment by MarkLieffering** [comment]
again? most of the PDF links return a Server Error 404 - File or directory not found.

**Comment by hol430** [comment]
Yes, I think #5258 fixed the problem, however it was a "working on" pull request so it didn't cause the website to be updated, and no "resolves" pull request has been merged since then. I have applied a temporary fix to make those files available, and hopefully it should sort itself out for good once a "resolves" pull request gets merged.

**Comment by hol353** [comment]
The auto-doc is failing on Jenkins when creating the wheat doc - see Jenkins logs. The CreateInstallation Jenkins job is green (blue actually) though which even though it failed.

**Comment by hol430** [comment]
@hol353 the reason that jenkins ran without errors but didn't generate a release is because the pull request didn't resolve an issue. I recently changed the APSIM.Builds code so that the CreateInstallation jobs don't get run unless the PR resolves an issue, so I'm not sure why this job even ran in the first place. Have you published to the APSIM.Builds webservice recently without pulling from master?

**Comment by hol353** [comment]
Nope I haven't published the APSIM.Builds webservice in a very long time.
